Chessington.GameEngine/Pieces/Rook.cs

```diff
@@ -10,7 +10,77 @@ public Rook(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            List<Square> availableMoves = new List<Square>();
+
+            for (int i = board.FindPiece(this).Col + 1; i < 8; i++)
+            {
+                if (board.GetPiece(new Square(board.FindPiece(this).Row, i)) == null)
+                {
+                    availableMoves.Add(new Square(board.FindPiece(this).Row, i));    
+                }
+                else if (board.GetPiece(new Square(board.FindPiece(this).Row, i)).Player != this.Player)
+                {
+                    availableMoves.Add(new Square(board.FindPiece(this).Row, i));
+                    break;
+                }
+                else
+                {
+                    break;
+                }
+            }
+            
+            for (int i = board.FindPiece(this).Col - 1; i >= 0; i--)
+            {
+                if (board.GetPiece(new Square(board.FindPiece(this).Row, i)) == null)
+                {
+                    availableMoves.Add(new Square(board.FindPiece(this).Row, i));    
+                }
+                else if (board.GetPiece(new Square(board.FindPiece(this).Row, i)).Player != this.Player)
+                {
+                    availableMoves.Add(new Square(board.FindPiece(this).Row, i));
+                    break;
+                }
+                else
+                {
+                    break;
+                }
+            }
+            
+            for (int i = board.FindPiece(this).Row + 1; i < 8; i++)
+            {
+                if (board.GetPiece(new Square(i, board.FindPiece(this).Col)) == null)
+                {
+                    availableMoves.Add(new Square(i, board.FindPiece(this).Col));    
+                }
+                else if (board.GetPiece(new Square(i, board.FindPiece(this).Col)).Player != this.Player)
+                {
+                    availableMoves.Add(new Square(i, board.FindPiece(this).Col));
+                    break;
+                }
+                else
+                {
+                    break;
+                }
+            }
+            
+            for (int i = board.FindPiece(this).Row - 1; i >= 0; i--)
+            {
+                if (board.GetPiece(new Square(i, board.FindPiece(this).Col)) == null)
+                {
+                    availableMoves.Add(new Square(i, board.FindPiece(this).Col));    
+                }
+                else if (board.GetPiece(new Square(i, board.FindPiece(this).Col)).Player != this.Player)
+                {
+                    availableMoves.Add(new Square(i, board.FindPiece(this).Col));
+                    break;
+                }
+                else
+                {
+                    break;
+                }
+            }
```

Comment:
In order to avoid repeating code you could make a function that checks the available squares in a specified direction eg. List<Squares> GoInDirection(Square startingPosition, int colDir, int rowDir). For the Rook to check all the right squares the colDir = 1 and rowDir = 0.